### PR TITLE
DashboardSettings: Link labels to inputs

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR links labels to its input
**Which issue(s) this PR fixes**:

Relates to #39841

### CODE REVIEW

**Comment by thisisobate** [comment]
RadioButtons seems to be quite tricky to link. This is as a result of how RadioButtons are constructed under the hood.

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 1]
```diff
@@ -67,14 +67,20 @@ export class TimePickerSettings extends PureComponent<Props, State> {
       <CollapsableSection label="Time options" isOpen={true}>
         <Field label="Timezone" aria-label={selectors.components.TimeZonePicker.container}>
           <TimeZonePicker
+            inputId="Time-options-input"
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:70]: nit: this should probably be lower case, `time-options-input`?

**Review by ashharrison90 — APPROVED** [review]
minor nit, lgtm otherwise 🙌

**Comment by thisisobate** [comment]
> RadioButtons seems to be quite tricky to link

Came up with a solution; would open a separate PR for this

**Pull request merged**